pkg/inttest/inttest.go

```diff
@@ -0,0 +1,315 @@
+// Package inttest enables writing of integration tests. Setup Docker containers for dependencies
+// like PostgreSQL, RabbitMQ and AWS S3 (using localstack). Every setup function ensures the
+// container is ready before returning, ensures resources are cleaned up after the tests are
+// finished and return a client ready to interact with the container.
+package inttest
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/go-redis/redis"
+	gnomockRedis "github.com/orlangure/gnomock/preset/redis"
+
+	amqp "github.com/rabbitmq/amqp091-go"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/dhis2-sre/im-manager/internal/handler"
+	"github.com/dhis2-sre/im-manager/internal/server"
+	"github.com/dhis2-sre/im-manager/pkg/storage"
+	"github.com/gin-gonic/gin"
+	_ "github.com/lib/pq" // postgres driver
+	"github.com/orlangure/gnomock"
+	"github.com/orlangure/gnomock/preset/localstack"
+
+	"github.com/dhis2-sre/im-manager/pkg/config"
+	"github.com/orlangure/gnomock/preset/postgres"
+	"github.com/orlangure/gnomock/preset/rabbitmq"
+
+	"github.com/stretchr/testify/require"
+	"gorm.io/gorm"
+)
+
+func SetupRedis(t *testing.T) *redis.Client {
+	container, err := gnomock.Start(gnomockRedis.Preset())
+	require.NoError(t, err, "failed to start Redis")
+	t.Cleanup(func() { require.NoError(t, gnomock.Stop(container), "failed to stop Redis") })
+
+	client := &redis.Options{
+		Addr:     container.DefaultAddress(),
+		Password: "",
+		DB:       0,
+	}
+	return redis.NewClient(client)
+}
+
+// SetupDB creates a PostgreSQL container. Gorm is connected to the DB and runs the migrations.
+func SetupDB(t *testing.T) *gorm.DB {
```

Comment:
How about renaming this to SetupPostgresql? Our other setup functions are... Vendor specific :smile: